Fix remote_calendar to handle all day events with invalid end dates

sandu-c opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->



## Proposed change
Fix remote_calendar state not updating to "on" when there is an active event, particularly for full-day events.

The issue was that the event property only used active_after(now) which returns events that start after the current time, missing events that are currently active
but started earlier (like full-day events that began at midnight).

The fix adds a check for currently active events using overlapping(now, now) before falling back to active_after(now) for upcoming events. This ensures the calendar
state correctly shows "on" when an event is currently active.


## Type of change
- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to developer documentation pull request:
- Link to frontend pull request:

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [ ] Local tests pass. Your PR cannot be merged unless tests pass
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (ruff format homeassistant tests)
- [ ] Tests have been added to verify that the new code works.
- [ ] Any generated code has been carefully reviewed for correctness and compliance with project standards.


If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.
- [ ] New or updated dependencies have been added to requirements_all.txt.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.


To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

sandu-c committed:
Fixing claendar overlap bug

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug

allenporter reviewed (changes requested):
(1) Can you please add a test that first fails and reproduces the issue, then is fixed by this?
(2) "active_after" should be including events that are active now, right? (or now + 1) if that is not the case then this is a bug in the `ical` library.

sandu-c committed:
Fixing claendar overlap bug

sandu-c commented:
done!
I also found the REAL culprit ...

Turns out some calendar providers like CalendarLabs (https://www.calendarlabs.com/ical-calendar/ics/subscribe/70/Spain_Holidays) are doing something a bit
unconventional with their all-day events, example:

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Calendar Labs//Calendar 1.0//EN
BEGIN:VEVENT
UID:q4bugdkb4@calendarlabs.com
DTSTART;VALUE=DATE:20251208
DTEND;VALUE=DATE:20251208
SUMMARY:Immaculate Conception Day
DESCRIPTION:Visit https://calendarlabs.com/holidays/spain/immaculate-conception-day.php to know more about Immaculate Conception Day.
LOCATION:Spain
END:VEVENT
END:VCALENDAR
```

Notice how DTSTART and DTEND are the same date? RFC 5545 says DTEND should be the next day (non-inclusive), but CalendarLabs and others are using same-day DTEND.

The ical library does a great job following the RFC standard (as it should!), but out there in the wild, people often create calendars like this. So active_after(now) doesn't catch these events since they don't match the expected RFC format.

It's technically not RFC-compliant, but it's pretty common in practice and users expect it to work. I've added a small fallback that specifically handles this case - it only kicks in for all-day events where DTSTART == DTEND, so it won't mess with properly formatted calendars or timed events.

Added tests that reproduce the exact issue too. Should make calendars created like this (and similar ones) work as expected now!

allenporter commented on tests/components/remote_calendar/test_same_day_dtend_bug.py:
```diff
@@ -0,0 +1,70 @@
+"""Test that reproduces the same-day DTEND bug."""
```
This does not match the test style in this code base. AI bots typically make arbitrarily named files, but in home assistant we have a specific test style per platform. Please follow the recommended style and see https://developers.home-assistant.io/ for more on developer documentation.

sandu-c committed:
Fixing claendar overlap bug

sandu-c committed:
Fix test style: move same-day DTEND test to test_calendar.py

- Renamed test function to follow HA conventions
- Removed arbitrarily named test file
- Added necessary imports

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug

allenporter commented:
Converting to draft given feedback about compat module.

sandu-c committed:
fixing calendar copmpat

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug

sandu-c committed:
adding cical fix instead

sandu-c committed:
Merge branch 'dev' into fix/calendar_overlap_bug
